### DISCUSSION TITLE
Render MIDI tracks to audio

### DISCUSSION THREAD

**Discussion by petersampsonaudacity** [audacity/audacity#893] [Ideas & Feature Requests]
What users can do with MIDI files imported into Note Tracks is extremely limited. Audacity is not a MIDI editor.

What Audacity is, is an audio editor. Thus it would be extremely useful if Audacity could Mix & Render Note Tacks to Audio tracks, then users could at least edit the audio and mix the rendered MIDI tracks with their normal audio tracks.

There have been many calls for this on the Forum over the years.. 

This is present on the Wiki as a Proposal:
https://wiki.audacityteam.org/wiki/Proposal:_Render_MIDI_tracks_to_audio

**Comment by pljones** [comment]
> What Audacity is, is an audio editor. Thus it would be extremely useful if Audacity could Mix & Render Note Tacks to Audio tracks

Isn't that actually a contradiction?  Yes, Audacity is an Audio editor.  Thus it makes no sense for it to be able to parse MIDI command sequences, synchronise interpretation of those commands to the audio time line, have a built in synthesiser to control using those commands and merge the synthesised audio with other tracks.  Because that's not what being an Audio editor is -- it's what being a MIDI Sequencer and Synthesiser is.

So what you're asking for is just that -- adding a MIDI Sequencer and a MIDI Synthesiser.  You'd probably want the MIDI Sequencer to run in time with the Audio time line (not so hard in music editing mode).  And you'd probably want to be able to replace any built-in MIDI Synthesiser with a plug-in (like MuseScore can).  Speaking of MuseScore... it's a MIDI Sequencer with a MIDI Synthesiser built in and you can replace that with your choice of plug-in.  (It's also a MIDI Editor, too, as a free bonus.)  It can also render the playback to a WAV (or other) audio file that you can then edit in the audio editor of your choice.  (Yes, it's a bit long winded maybe, but it's going to get you a result quickly and sounding as good as the quality of the plugins you choose to use.)

So you're effectively asking for Audacity and MuseScore to become one program.  Or you could use a full Digital Audio Workstation like [Reaper](https://www.reaper.fm) (I prefer editing audio in Audacity still, despite recent changes making the UI worse, and editing MIDI in MuseScore is more natural than in Reaper).
